schemas/v3.1/schema.yaml

```diff
@@ -474,10 +474,24 @@ $defs:
     type: object
     additionalProperties:
       $ref: '#/$defs/media-type'
+    patternProperties:
+      '^application/x-www-form-urlencoded$|^multipart\/.+$':
+        $ref: '#/$defs/media-type-with-encoding'
     propertyNames:
       format: media-range
 
   media-type:
+    $comment: https://spec.openapis.org/oas/v3.1.0#media-type-object
+    type: object
+    properties:
+      schema:
+        $dynamicRef: '#meta'
+    allOf:
+      - $ref: '#/$defs/specification-extensions'
+      - $ref: '#/$defs/examples'
+    unevaluatedProperties: false
+
+  media-type-with-encoding:
```

Comment:
I would factor out everything from `media-type` _except_ `unevaluatedProperties` into a `media-type-common`, and then `$ref: "#/$defs/media-type-common"` from both `media-type` and `media-type-with-encoding`.  Both would then have `unevaluatedProperties: false` and `media-type-with-encoding` would have a `properties` keyword adding the `encoding` property, but no duplication of `type` or the other properties or `allOf`s.

Reply:
do you mean like this? 

```yaml
media-type-common:
   $comment: https://spec.openapis.org/oas/v3.1.0#media-type-object
    type: object
    properties:
      schema:
        $dynamicRef: '#meta'
    allOf:
      - $ref: '#/$defs/specification-extensions'
      - $ref: '#/$defs/examples'

 media-type:
   $comment: https://spec.openapis.org/oas/v3.1.0#media-type-object
   $ref: '#/$defs/media-type-common'
   unevaluatedProperties: false 

media-type-with-encoding:
   $comment: https://spec.openapis.org/oas/v3.1.0#media-type-object
    $ref: '#/$defs/media-type-common'
    type: object
    properties:
      encoding:
        $ref: '#/$defs/encoding-object'
    unevaluatedProperties: false
```

Reply:
I did consider something like this just to avoid updating media-type in two places.

Reply:
Yup, like that!  If we have similar places where this kind of `$comment` gets replicated, then definitely replicate it like you are now.  Otherwise I'd be willing to bet that you can just put it in the common schema.  Although I'm not sure if anything is relying on them - you're not supposed to programmatically rely on `$comment` in general.